tests/test_converter_cqasm.py

```diff
@@ -0,0 +1,255 @@
+# MIT License
+#
+# Copyright (c) 2022 Quandela
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import pytest
+
+from pathlib import Path
+
+from perceval import StateVector, pdisplay
+from perceval.converters import CQASMConverter, ConversionSyntaxError, ConversionUnsupportedFeatureError
+import perceval.components.unitary_components as components
+from perceval.components import catalog
+from perceval.rendering.format import Format
+
+import numpy as np
+
+
+def test_converter_syntax_error():
+    source = """
+version 3
+rabbit r
+"""
+    with pytest.raises(ConversionSyntaxError):
+        CQASMConverter(catalog).convert_string(source)
+
+
+def test_converter_unsupported_classical_variable():
+    source = """
+version 3
+qubit[2] q
+bit b
+X q[0]
+"""
+    with pytest.raises(ConversionUnsupportedFeatureError):
+        CQASMConverter(catalog).convert_string(source)
+
+
+def test_converter_unsupported_gates():
+    source = """
```

Comment:
Same comment for the variable name `source`. Please edit other instances (below) too in this file.

Reply:
Great!

Thanks for the review, please check the latests commits.